### ISSUE TITLE
Kiali SA should use view-only role unless using anonymous strategy

### ISSUE THREAD

**Issue by technotaff-nbs** [kiali/kiali#5611]
## Feature Request
Currently the following setting in the kiali-server helm chart:

```
deployment:
  view_only_mode: true
```
performs the following actions:
1. Caused the kiali-viewer ClusterRole to be bound the the kiali service-account.
2. Causes the kiali-server front-end to operate in view-only mode (even when logged in with a token that has edit privileges)

Would it be possible to split this functionality, so that the ClusterRole assignment is done on a *different* values-file, e.g.:

```
deployment:
  view_only_mode: false
  sa_view_only: true
```
In this scenario:
1. Caused the kiali-viewer ClusterRole to be bound the the kiali service-account.
2. Causes the kiali-server front-end to operate in edit mode if the logged-in user has the correct rights (say, in a cluster-admin role).

## Reasoning
We operate in a secure environment and try to provide least-privilege to all service-accounts. However when we do this with kiali , properly-authenticated users are unable to edit istio resources from the UI.

**Comment by jmazzitelli** [comment]
I do not believe that will not be possible because some of under-the-cover work done by Kiali is done via the SA (not everything is done via the user credentials). I suspect this kind of mixing of permissions might break things. I say this but I am not 100% sure in my answer though. Feel free to perform some investigation if its possible to turn on "view-only" in the SA but have users be able to edit resources through the Kiali frontend (would need some hacking of the frontend code to do things like show the "Edit" buttons in the editors even if view-only it enabled).

**Comment by technotaff-nbs** [comment]
We have patched the ClusterRoleBinding and restarted Kiali, everything *seems* okay. I'll double check for errors in the logs.

**Comment by Jinn22** [comment]
I don't see any errors in kiali pods when kiali service account is bound to kiali-viewer ClusterRole while I log into Kiali using a token with privileged access and I have admin access in Kiali UI.

**Comment by jmazzitelli** [comment]
This is valid use case for the operator; not just helm chart. Updated the title.

Now that I think about it - assuming it all works - there is no need for the Kiali SA to have read-write privileges if Kiali will use the user's credentials/token (and that is the case when using auth.strategy of "token", "openid", and "openshift" (and I believe "header" too). The only time Kiali SA would need read-write is if using auth.strategy of anonymous and view_only_mode is off.

So whoever works on this enhancement should (a) confirm that truly kiali SA does NOT need read-write perms at all when using a non-anonymous auth.strategy and (b) make this change for operator and helm charts both. This would require substantial changes to the role templates themselves, but I doubt much else will need to change.

**Comment by jmazzitelli** [comment]
I'm going to see if we can do this work. In fact, I think what we will do is make the Kiali SA "view-only" all the time - EXCEPT when using auth.strategy=anonymous. When auth.strategy == anonymous, the Kiali SA is what the user will be using to determine permissions. In this case, the Kiali SA will be given read-write permissions EXCEPT when the view-only setting is enabled - in that case, we'll keep the Kiali SA as "view only".

So, in short:

1. The Kiali SA will, in almost every case, be set to limited permissions (i.e. the view-only permissions)
2. The only time Kiali SA will be given full read-write permissions is if (a) auth.strategy is `anonymous` and (b) deployment.view_only_mode is set to `false`

This work will not be done this week for the upcoming release, but perhaps we can get this done in the following sprint/release.

**Comment by jmazzitelli** [comment]
PRs in progress:
* https://github.com/kiali/helm-charts/pull/180
* https://github.com/kiali/kiali-operator/pull/590

**Comment by jmazzitelli** [comment]
We can merge the PRs after someone reviews.

I will change the title of this issue - we won't split the functionality but what we will do is always set the Kiali SA to use the "view_only" Role UNLESS auth.strategy is "anonymous". But it seems the Kiali SA never needs read-write permissions in non-anonymous mode so that's what will be used, thus no need for the user to use a separate setting as originally described in the original enhancement request.
